### ISSUE TITLE
Fix Magento version comparison + version number color bug?

### ISSUE THREAD

**Issue by grafikchaos** [davidalger/capistrano-magento2#64]
Unable to complete the deployment due to this error:

```bash
ArgumentError: Malformed version number string 2.1.5
```

Tracing it back to [`magento2.rb`](https://github.com/davidalger/capistrano-magento2/blob/master/lib/capistrano/magento2.rb#L16) and that maybe the `capture :php` portion of the code isn't returning a string as expected?

```ruby

SSHKit.config.command_map[:magento] = "/usr/bin/env php -f bin/magento --"

module Capistrano
  module Magento2
    module Helpers
      def magento_version
        return Gem::Version::new((capture :php, "-f #{release_path}/bin/magento -- -V").split(' ').pop)
      end

      # ...more methods...
  end
end

```

Looking through the various forks I found [minhnd55/capistrano-magento2 has a solution that works ](https://github.com/minhnd55/capistrano-magento2/commit/1d0e187c5ec0e8c146c41dbdf052df6871f03c30)

I'll have a pull request ready soon to patch the issue.

**Comment by davidalger** [comment]
@grafikchaos Thanks for the detailed error report! To work on this, I'm going to need some more information if you're able to get it for me.

Can you tell me what OS you're running on the target machine, what version of php you're using, what version of ruby is running on the machine you're deploying from and also send me the contents of `log/capistrano.log` around the version call?

Also, can you run the following command on the same machine you are deploying from and tell me what it outputs?

    ruby -e 'puts Gem::Version::new("2.1.5")'

I can't reproduce the issue, and haven't seen it on any of our deployments, so I'm hoping this will help me get somewhere. I'm also a bit perplexed with the regex `\e\[(\d+)m` as it is matching on a shell escape sequence, but it's not accounting for the period in the version number, which means the version number it returns is going to be inaccurate. So I need to figure out first how to reproduce the issue so I can verify a solution.

**Comment by grafikchaos** [comment]
The only thing that seems to make sense is a hidden character that formats the version number in yellow (see screenshot). Every time I tried debugging the version number in a ruby shell I never had a problem, only when it was dynamically captured with colored output.

![screen shot 2017-03-29 at 2 30 49 pm](https://cloud.githubusercontent.com/assets/29170/24472729/62bd7e1e-148c-11e7-84bf-d1b2421793dd.png)

> Also, can you run the following command on the same machine you are deploying from and tell me what it outputs?

```zsh
ruby -e 'puts Gem::Version::new("2.1.5")'
2.1.5
```


## Remote target environment:

+ **os:** CentOS release 6.8 (Final)
+ **php:** PHP 7.0.15
+ **shell:** `bash`


```bash
PHP 7.0.15 (cli) (built: Jan 23 2017 15:57:44) ( NTS )
Copyright (c) 1997-2017 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2017 Zend Technologies
    with Zend OPcache v7.0.15, Copyright (c) 1999-2017, by Zend Technologies
```

## Local machine for deployments:

+ **os:** macOS Sierra 10.12.3
+ **ruby:** ruby 2.1.3p242 (2014-09-19 revision 47630) [x86_64-darwin16.0] (installed via rbenv)
+ **shell:** `zsh`

### Local Gems

**Gemfile**
```ruby
# frozen_string_literal: true
source "https://rubygems.org"

gem 'capistrano-magento2', "0.6.0"# github: 'augustash/capistrano-magento2', branch: "hotfix/pr64-fix-version-comparison"
gem "airbrussh", :require => false
```

**Installed gems**
```bash
gem list -d airbrussh                                                                                         develop

*** LOCAL GEMS ***

airbrussh (1.1.2, 1.1.1, 1.0.0, 0.8.0)
    Author: Matt Brictson
    Homepage: https://github.com/mattbrictson/airbrussh
    License: MIT
    Installed at (1.1.2): /usr/local/opt/rbenv/versions/2.1.3/lib/ruby/gems/2.1.0
                 (1.1.1): /usr/local/opt/rbenv/versions/2.1.3/lib/ruby/gems/2.1.0
                 (1.0.0): /usr/local/opt/rbenv/versions/2.1.3/lib/ruby/gems/2.1.0
                 (0.8.0): /usr/local/opt/rbenv/versions/2.1.3/lib/ruby/gems/2.1.0

    Airbrussh pretties up your SSHKit and Capistrano output
```



## capistrano.log
![screen shot 2017-03-29 at 2 53 16 pm](https://cloud.githubusercontent.com/assets/29170/24473657/a320a708-148f-11e7-804c-668875d7ec93.png)

```
** DEPLOY FAILED
** Refer to log/capistrano.log for details. Here are the last 20 lines:


 DEBUG [db1deae7] Running /usr/bin/env php -f /home/mastersu/master-supplements.com/staging/releases/20170329195018/bin/magento -- -V as mastersu@208.69.121.16

 DEBUG [db1deae7] Command: cd /home/mastersu/master-supplements.com/staging/releases/20170329195018 && ( export PATH="/opt/nexcess/php70u/root/usr/bin/:$PATH" ; /usr/bin/env php -f /home/mastersu/master-supplements.com/staging/releases/20170329195018/bin/magento -- -V )

 DEBUG [db1deae7] 	Magento CLI version 2.1.5
```

**Comment by davidalger** [comment]
Thank you so much for the completeness of your response! 

>The only thing that seems to make sense is a hidden character that formats the version number in yellow (see screenshot).

You are spot on with this! Crazy thing is, ansi output shouldn't turn up in non-interactive shells, but I'm pretty sure that's what's causing this.

I've pushed a commit to develop which should solve this. In short, I added `--no-ansi` to the `bin/magento` call. You can see the difference it makes:
![current ssh www-stage stage pksafety com bash 127x45 2017-03-30 12-26-04](https://cloud.githubusercontent.com/assets/658281/24525056/89662598-155e-11e7-8ba6-553eb8a7f6fc.png)

If you can test this fix and confirm that it solves the issue, I'll tag a new release and push it up ASAP.

**Comment by davidalger** [comment]
I went ahead and released version 0.6.1 which has my fix. If the issue persists, feel free to re-open the task so we can get a better solution in place. Thanks again for the report!
